src/thor/triplegbuilder.cc

```diff
@@ -1251,7 +1245,8 @@ TripLegBuilder::Build(const AttributesController& controller,
     // Assign the trip path admins
     AssignAdmins(controller, trip_path, admin_info_list);
 
-    return trip_path;
+    // Trivial path is done
+    return;
```

Comment:
noticed this as well. we were taking the trip leg as a reference, modifying it and then returning a copy of it... basically we did a refactor and forgot to do it all the way. this will be more efficient now :)